app/assets/javascripts/components/story/CollapsedStory/index.js

```diff
@@ -25,65 +29,187 @@ const storyClassName = (story, additionalClassname = '') => {
       'Story--estimated': !isStoryNotEstimated
     },
     additionalClassname
-  );
-};
+  )
+}
 
-export const CollapsedStory = ({ 
-  onToggle, 
-  story, 
-  updateCollapsedStory, 
-  project, 
-  className, 
-  title, 
-  highlight, 
-  isHighlightable 
-}) =>
-  <div
-    className={storyClassName(story, className)}
-    onClick={onToggle}
-    title={title}
-  >
-    <StoryPopover story={story}>
-      <div className='Story__icons-block'>
-        <StoryIcon storyType={story.storyType} />
-        <CollapsedStoryEstimate estimate={story.estimate} />
-        <StoryDescriptionIcon description={story.description} />
-      </div>
-    </StoryPopover>
-
-    <CollapsedStoryInfo story={story} />
-
-    <CollapsedStoryStateActions story={story}
-      onUpdate={(newAttributes) =>
-        updateCollapsedStory(story.id, project.id, newAttributes)}
-    />
-    {
-      isHighlightable && 
-      <CollapsedStoryFocusButon onClick={() => highlight(story.id)} />
+const type = {
+  story: 'STORY'
+}
+
+const calculatePosition = (aboveStory, bellowStory, storiesArray, index) => {
```

Comment:
We could create a new model file. Like `dragAndDrop.js` and extract all this functions for there.

Reply:
I agree completely that file is too dense right now.